source/common/http/http2/codec_impl.h

```diff
@@ -265,11 +266,15 @@ class ClientConnectionImpl : public ClientConnection, public ConnectionImpl {
   // Propogate network connection watermark events to each stream on the connection.
   // The router will propogate it downstream.
   void onUnderlyingConnectionAboveWriteBufferHighWatermark() override {
```

Comment:
Just to close the loop on this: What do you think about having the onUnderlying* functions for both server side and client side (so just move to base connection). This would unify the logic here vs. having part of it here and part of it in connection manager? I feel like that is easier to understand? No problem if you want to do that in different PR, but figured it's worth discussing since it impacts this change.